src/breakupModel/simulation/Breakup.cpp

```diff
@@ -1,110 +1,101 @@
 #include "Breakup.h"
 
 void Breakup::run() {
-    //1. Step: Remove all old stuff which there possible is and makes the Breakup ready to go
-    this->prepare();
+    //1. Step: Generate the new Satellites
+    this->calculateFragmentCount();
 
-    //2. Step: Generate the new Satellites
-    this->generateFragments();
-
-    //3. Step: Assign every new Satellite a value for L_c
+    //2. Step: Assign every new Satellite a value for L_c
     this->characteristicLengthDistribution();
 
-    //4. Step: Calculate the A/M (area-to-mass-ratio), A (area) and M (mass) values for every Satellite
+    //3. Step: Calculate the A/M (area-to-mass-ratio), A (area) and M (mass) values for every Satellite
     this->areaToMassRatioDistribution();
 
-    //5. Step: Assign parent and by doing that assign each fragment a base velocity
+    //4. Step: Assign parent and by doing that assign each fragment a base velocity
     this->assignParentProperties();
 
-    //6. Step: Calculate the Ejection velocity for every Satellite
+    //5. Step: Calculate the Ejection velocity for every Satellite
     this->deltaVelocityDistribution();
 
+    //6. Step: As a last step set the _currentMaxGivenID to the new valid value
+    _currentMaxGivenID += _output.size();
 }
 
-Breakup &Breakup::setSeed(unsigned long seed) {
-    _randomNumberGenerator = std::mt19937{seed};
+Breakup &Breakup::setSeed(std::optional<unsigned long> seed) {
+    if (seed.has_value()) {
+        _fixRNG = std::mt19937 {seed.value()};
+    } else {
+        _fixRNG = std::nullopt;
+    }
     return *this;
 }
 
-void Breakup::prepare() {
-    _output.clear();
-}
-
 void
-Breakup::createFragments(size_t fragmentCount, const std::array<double, 3> &position) {
-    _output.resize(fragmentCount, Satellite(SatType::DEBRIS, position));
-    std::for_each(_output.begin(), _output.end(),
-                  [&](Satellite &sat) { sat.setId(++_currentMaxGivenID); });
+Breakup::generateFragments(size_t fragmentCount, const std::array<double, 3> &position) {
+    //Just in case of a rerun - reset the outputMass to zero
```

Comment:
Just to be sure: Are there any other state variables or similar which ought to be reset? If so, should there be a "resetState" function or similar? Alternatively / additionally, should there be a warning if it is called again with potentially same state?

Reply:
Good that you reminded me.
``The _outpuMass = 0`` here is to prevent that nothing bad happens if someone modifies the Breakup. Normally this should not be required.

In a later iteration, I added a ``init()`` method to improve this issue. I've moved the ``outputMass = 0`` into this method.